Update expired lock test for 0.50.0 behavior

tests.object.test_object_lock.TestObjectLockWithGrpc#test_already_outdated_lock_should_not_be_applied 

AssertionError: Regex pattern did not match.
 Regex: 'lock object expiration: 47; current: 48'
 Input: "Command: ./neofs-cli --config /home/runner/work/neofs-node/neofs-node/neofs-testcases/wallet_config.yml object lock --rpc-endpoint 'localhost:40233' --wallet '/home/runner/work/neofs-node/neofs-node/neofs-testcases/test-run-2025-11-20-09-52-01-468649/wallet-1d165b18-8323-4ded-b648-23701a7504c6.json' --cid 'Fhx8ZAkMgpUvjqJ3TkwymVsaQjiGN3b96bjSF5NWNqLu' --oid '69Bo4otoMgBSsbij7uDvN6Yqf8NpGVAqxjzUKsmDE3ea' --expire-at 47\nError:\nreturn code: 1\noutput: Error: store lock object for 69Bo4otoMgBSsbij7uDvN6Yqf8NpGVAqxjzUKsmDE3ea in NeoFS: finish object stream: status: code = 1024 message = object has expired\n"

https://rest.fs.neo.org/Aku38wpn2xEqfEPxmwyAa7N3VnAkMV8PEVkpLqwKM1z3/4348-1763637514/index.html#suites/295ed2e000f8fa6f3ade59cc13b98615/b0db26dd15346cc8/

It's the same error, but returned a bit differently after the https://github.com/nspcc-dev/neofs-node/pull/3684. Both errors are correct, test needs to be adjusted for it.

--- a/pytest_tests/lib/helpers/grpc_responses.py
+++ b/pytest_tests/lib/helpers/grpc_responses.py
@@ -29,7 +29,7 @@ OBJECT_HEADER_LENGTH_LIMIT = ".*object header length exceeds the limit.*"
 
 LIFETIME_REQUIRED = ".*at least one of the flags in the group [expire-at lifetime] is required.*"
 LOCK_OBJECT_REMOVAL = "lock object removal"
-LOCK_OBJECT_EXPIRATION = "lock object expiration: {expiration_epoch}; current: {current_epoch}"
+LOCK_OBJECT_EXPIRATION = "(lock object expiration: {expiration_epoch}; current: {current_epoch}|object has expired)"
 INVALID_RANGE_ZERO_LENGTH = "invalid '{range}' range: zero length"
 INVALID_RANGE_OVERFLOW = "invalid '{range}' range: uint64 overflow"
 INVALID_OFFSET_SPECIFIER = "invalid '{range}' range offset specifier"

